Finalized 1.0 release

glbrtmrgn opened the pull request:
### Acceptance Criteria

- [x] Write more complete integration tests.
    - [x] `tests/generate-test.js`
    - [x] `tests/render-test.js`
    - [x] `tests/data-test.js`
- [ ] Document how to  publish releases to npm, [using the npm way](https://docs.npmjs.com/getting-started/publishing-npm-packages).
- [x] Include an [`.eslintrc`](http://eslint.org/) and [`.editorconfig`](http://editorconfig.org/), the latter with a JS focus.
- [x] Refactor render code to be a `lib/render.js` file instead

glbrtmrgn committed:
Fixed the preinstall issue

We separated reinstall into two hooks(preinstall and postinstall)

glbrtmrgn committed:
Abstracted the render functionality

Made render into its own function and created a simple test for this.

glbrtmrgn committed:
Refactored data

We moved the message property from generate to data.

glbrtmrgn committed:
Adding test for the generate library

Integration testing is now complete

rogeruiz committed:
Add Travis CI support for integration tests

rogeruiz committed:
Skip plugin installation if TRAVIS

rogeruiz committed:
Better expression

rogeruiz committed:
Not exiting with a nonzero status

rogeruiz committed:
Add `addons` stanza because the internet showed me

rogeruiz committed:
Add some docs around Travis CI

glbrtmrgn committed:
Merge pull request #6 from 18F/add-ci

Add Travis CI support for integration tests

rogeruiz commented on README.md:
```diff
@@ -54,6 +56,12 @@ is easily made by answering the following question.
+This project is tested with Travis-CI. The current versions of node that are
+being tested are the `latest`, `6.x`, and `5.x`. Please see the `.travis.yml` at
+the root of this repository for more info.
+
```
Might make some sense to talk about how `master` is the default branch for the build status badge in here before merging into `master`. 💭

rogeruiz committed:
Update deployment to a script

rogeruiz committed:
Add better documentation;

I've added some deployment documentation that might need a run through
and some writing prompts for #g-frontend contributions to help teams
make Coming Attractions in a repeatable way that might be able to stay
in sync with their current app.

rogeruiz committed:
Welcome aboard, @glbrtmrgn :wave:

rogeruiz committed:
Update docs

glbrtmrgn committed:
We added linting! :tada:

glbrtmrgn committed:
Ensured that we run lint when we test

glbrtmrgn committed:
Added the editorconfig file

glbrtmrgn committed:
Merge pull request #7 from 18F/update-docs

Update docs

rogeruiz reviewed (approved):
:shipit:
